experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -0,0 +1,540 @@
+import ClimaParams
+import ClimaLand as CL
+import Dates
+import ClimaUtilities.TimeVaryingInputs: LinearInterpolation, PeriodicCalendar, TimeVaryingInput
+import ClimaCoupler: Checkpointer, FieldExchanger, FluxCalculator, Interfacer, Utilities
+import ClimaCore as CC
+import SciMLBase
+import ClimaTimeSteppers as CTS
+import ClimaDiagnostics as CD
+import ClimaUtilities.TimeManager: ITime
+
+include("climaland_helpers.jl")
+
+"""
+    ClimaLandSimulation{M, P, D, I, A}
+
+The integrated ClimaLand model simulation object.
+"""
+struct ClimaLandSimulation{M, D, I, A} <: Interfacer.LandModelSimulation
+    model::M
+    domain::D
+    integrator::I
+    area_fraction::A
+end
+
+"""
+    ClimaLandSimulation(
+        ::Type{FT},
+        dt::TT,
+        tspan::Tuple{TT, TT},
+        start_date::Dates.DateTime,
+        output_dir::String,
+        boundary_space,
+        area_fraction;
+        saveat::Vector{TT} = [tspan[1], tspan[2]],
+        domain_type::String = "sphere",
+        land_temperature_anomaly::String = "amip",
+        use_land_diagnostics::Bool = true,
+        stepper = CTS.ARS111(),
+    )
+
+Creates a ClimaLandSimulation object containing a land domain,
+a ClimaLand.LandModel, and an integrator.
+
+This type of model contains a canopy model, soil model, snow model, and
+soil CO2 model. Specific details about the complexity of the model
+can be found in the ClimaLand.jl documentation.
+"""
+function ClimaLandSimulation(
+    ::Type{FT},
+    dt::TT,
+    tspan::Tuple{TT, TT},
+    start_date::Dates.DateTime,
+    output_dir::String,
+    boundary_space,
+    area_fraction;
+    saveat::Vector{TT} = [tspan[1], tspan[2]],
+    domain_type::String = "sphere",
+    surface_elevation = CC.Fields.zeros(boundary_space),
+    land_temperature_anomaly::String = "amip",
+    use_land_diagnostics::Bool = true,
+    stepper = CTS.ARS111(),
+) where {FT, TT <: Union{Float64, ITime}}
+    @assert domain_type == "sphere" "Currently only spherical shell domains are supported; single column may be supported in the future."
+
+    # Set up domain
+    depth = FT(10) # soil depth
+    n_vertical_elements = 10
+    domain = make_land_domain(boundary_space, (-depth, FT(0.0)), n_vertical_elements)
+    surface_space = domain.space.surface
+    subsurface_space = domain.space.subsurface
+
+    # Set up spatially-varying parameters
+    earth_param_set = CL.Parameters.LandParameters(FT)
+    spatially_varying_soil_params = CL.default_spatially_varying_soil_parameters(subsurface_space, surface_space, FT)
```

Comment:
I think the land model should be providing an easier way to set this up! otherwise the standard land model and what is used in the coupler will rapidly be out of sync

Reply:
Agreed. We can work on it on the land side - I opened an issue here https://github.com/CliMA/ClimaLand.jl/issues/1067

Reply:
Thank you!